scripts/portal-flow.js

```diff
@@ -1,7 +1,89 @@
-module.exports = function(runner) {
+'use strict';
+const configureRequest = require('../util/configureRequest');
+const requestBodyUtils = require('../util/sync_request_bodies');
+const recordUtils = require('../util/generate_record');
+const Promise = require('bluebird');
+let baseUrl = '';
+let clientId = '';
+let request;
+
+function urlFor(dataset) {
+  return `${baseUrl}/mbaas/sync/${dataset}`;
+}
+
+function syncDataset(name) {
+  const payload = requestBodyUtils.getSyncRecordsRequestBody({
+    dataset_id: name,
+    meta_data: {
+      clientIdentifier: clientId
+    },
+    pending: []
+  });
+  return request.post({
+    url: urlFor(name),
+    body: payload,
+    json: true
+  });
+}
+
+function create(dataset, data) {
+  const payload = requestBodyUtils.getSyncRecordsRequestBody({
+    fn: 'sync',
+    meta_data: {
+      clientIdentifier: clientId
+    },
+    pending: [recordUtils.generateRecord(data)]
+  });
+  return request.post({
+    url: urlFor('workorders'),
+    body: payload,
+    json: true
+  });
+}
+
+module.exports = function(runner, argv) {
   return function(previousResolution) {
     runner.actStart('Portal Flow');
-    runner.actEnd('Portal Flow');
-    return Promise.resolve(previousResolution);
+    baseUrl = argv.app;
+    clientId = previousResolution.clientIdentifier;
+    request = configureRequest(clientId, previousResolution.sessionToken);
```

Comment:
Yeah, I was initially doing that, using variables this way hurts my feelings too, but the list of parameters for the functions were growing a bit too much...

This is kinda using `this.baseUrl` without the prototypes, I'm policing myself to only assign to them once, but I'll take another look at how this looks with them as parameters.